react/game/src/app/screens/Profile/components/ProfileEdit/components/ProfileEditForm/index.js

```diff
@@ -0,0 +1,129 @@
+          placeholder={placeholders.NAME}
+        />
+        <Field
+          type="text"
+          validate={[required]}
+          component={Input}
+          name={fields.lastName.name}
+          id={fields.lastName.id}
+          label={fields.lastName.label}
+          placeholder={placeholders.LASTNAME}
+        />
+        <Field
+          type="number"
+          validate={[required, minValueAge, maxValueAge]}
+          component={Input}
+          name={fields.age.name}
+          id={fields.age.id}
+          label={fields.age.label}
+          placeholder={placeholders.AGE}
+        />
+        <Field
+          validate={[required]}
+          component={Textarea}
+          id={fields.aboutMe.id}
+          name={fields.aboutMe.name}
+          label={fields.aboutMe.label}
+          placeholder={placeholders.ABOUT}
+          rows={5}
+          resize={false}
+        />
+        <Field
+          type="text"
+          validate={[required, url]}
+          component={Input}
+          name={fields.profilePicture.name}
+          id={fields.profilePicture.id}
+          label={fields.profilePicture.label}
+          placeholder={placeholders.URL}
+        />
+
+        <Field
+          type="text"
+          validate={[required, url]}
+          component={Input}
+          name={fields.backgroundPicture.name}
+          id={fields.backgroundPicture.id}
+          label={fields.backgroundPicture.label}
+          placeholder={placeholders.URL}
+        />
+
+        {successMessage && <div className={styles.successMessage}>{successMessage}</div>}
+        {errorMessage && <div className={styles.errorMessage}>Error: {errorMessage}</div>}
+        <div className={styles.editProfileBtnContainer}>
+          <Button
+            type="submit"
+            visible
+            disabled={pristine || submitting}
+            title={btnTooltips.save}
+            classes={styles.editProfileButtons}
+          >
+            {btnTitles.save}
+          </Button>
+          <Button
+            type="button"
+            onClick={activateEditProfile}
+            classes={`${styles.editProfileCancelBtn} ${styles.editProfileButtons}`}
+            visible
+            title={btnTooltips.cancel}
+          >
+            {btnTitles.cancel}
+          </Button>
+        </div>
+      </form>
+    </div>
+  );
+}
+
+ProfileEditForm.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  pristine: PropTypes.bool.isRequired,
+  submitting: PropTypes.bool.isRequired,
+  errorMessage: PropTypes.string,
+  successMessage: PropTypes.string,
+  activateEditProfile: PropTypes.func.isRequired
+};
+
+const mapStateToProps = state => ({
+  initialValues: state.profile.info,
+  errorMessage: state.profile.errorMessage,
+  successMessage: state.profile.successMessage
+});
+
+const ReduxFormProfileForm = reduxForm({
+  form: formNames.EDIT_PROFILE.name
+})(ProfileEditForm);
+
+export default connect(mapStateToProps)(ReduxFormProfileForm);
```

Comment:
Split this component to smart and dumb components

Reply:
okey dokey you are totally right!